src/batou/lib/file.py

```diff
@@ -18,15 +18,23 @@
 from batou.utils import dict_merge
 
 
-def ensure_path_nonexistent(path):
+def ensure_path_nonexistent(path: str) -> None:
+    """Ensure that the given path does not exist.
```

Comment:
Explain the logic that in a multi-user and concurrent system there isn't really a coarse grained guarantee around files not existing.

What this can guarantee, however, is that during the lifetime of this function call, there will have been an atomic moment where the file did not exist, or, if it existed and can not be removed, it will raise an error.

This means: if the file wasn't there in the first place, we're fine. If the file existed and we removed it, we are fine, we don't care whether someone else might have recreated it immediately after. If the file existed and was removed by someone else, we don't care either. If the file existed and we can't remove it, we need to fail.